mbbsd/bbs.c

```diff
@@ -433,6 +433,26 @@ int IsFreeBoardName(const char *brdname)
     return 0;
 }
 
+/*
+ * Check if user ip address is from Taiwan
+ *
+ * This function will compare from_cc string and "Taiwan" in Chinese (Big5)
+ * from_cc will generate when user login bbs system, and set from FORMD
+ *
+ * Return 1 when user is from Taiwan IP, 0 when outside Taiwan
+ */
+int is_taiwan_ip()
+{
+#ifdef POST_COMMENT_WITH_TAIWAN_IP_ONLY
```

Comment:
Might just wrap this `#ifdef` around the whole function, so you don't need the `#else` part.

Reply:
Let me think twice. If I warp whole function, I will need to change the function name or the future developer might use this function without `POST_COMMENT_WITH_TAIWAN_IP_ONLY` flag and got compiler error unexpected.

Reply:
That should be a compiler error, right?

I don't think it's correct that `is_taiwan_ip()` always return `true` without `POST_COMMENT_WITH_TAIWAN_IP_ONLY`.

You also have a ifdef in the header file already.

Reply:
Yes, you're right.